serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/integration-tests-with-postgresql.adoc

```diff
@@ -1 +1,192 @@
+
+[[Add-postgreSQL-test-resource]]
+== PostgreSQL instance as test resource
+The test uses PostgreSqlQuarkusTestResource to start the PostgresSQL instance.
+PostgreSqlQuarkusTestResource is a predefined test resource provided by kogito-quarkus-test-utils that starts a container with the PostgreSQL instance.
+That container has a full set of default configuration that are included in the integration test automatically
+
+[NOTE]
+====
+As a test resource, it will run its {quarkus_test_tesource_lifecycle_manager_class}[QuarkusTestResourceLifecycleManager] before the service test execution. More related info in {quarkus_testing_guide_test_resource}[Starting services before the Quarkus application starts].
+====
+
+The container has a full set of default configuration, such as PostgreSQL docker image name, default user, default password,...
+All the needed configuration values to link the Kogito Serverless Workflow service with that instance, are added automatically to the integration test.
+
+.Properties added automatically by PostgreSqlQuarkusTestResource
+[source, properties]
+----
+quarkus.datasource.reactive.url
+quarkus.datasource.jdbc.url
+quarkus.datasource.username
+quarkus.datasource.password
+----
+
+[[Add-conditional-postgreSQL-test-resource]]
+== Conditional PostgreSQL test resource
+
+This test resource can be disabled by a configuration like it's done in the `serverless-workflow-examples/serverless-workflow-newsletter-subscription` example.
+This allows you to reuse the same integration test with PostgreSQL enabled just for profiles with persistence enabled.
+In order to achieve this resource activation we have created this local test resource:
+
+.Conditional test resource creation
+[source,java]
+----
+@QuarkusIntegrationTest
+@QuarkusTestResource(SubscriptionResourceIT.ConditionalPostgreSqlQuarkusTestResource.class)
+public class SubscriptionResourceIT {
+    ...
+    public static class ConditionalPostgreSqlQuarkusTestResource extends PostgreSqlQuarkusTestResource {
+
+        public ConditionalPostgreSqlQuarkusTestResource() {
+            enableConditional();
+        }
+
+        private boolean isEnabled() {
+            return Optional.ofNullable(System.getProperty("enable.resource.postgresql")).map(property -> property.equalsIgnoreCase(Boolean.TRUE.toString())).orElse(false);
+        }
+
+        @Override
+        protected Map<String, String> getProperties() {
+            return isEnabled() ? super.getProperties() : new HashMap<>();
+        }
+    }
+}
+----
+
+This new resource ConditionalPostgreSqlQuarkusTestResource only will be started when the `enable.resource.postgresql` system property is true.
+In the example, the `persistence` profile incorporates this property setting to execute the integration test with the ConditionalPostgreSqlQuarkusTestResource enabled.
```

Comment:
```suggestion
In the example, the `persistence` profile incorporates this property setting to execute the integration test with the `ConditionalPostgreSqlQuarkusTestResource` enabled.
```

What do you mean by "property setting"?

Reply:
Maybe also explain why the Maven profile was added. See this example: https://kiegroup.github.io/kogito-docs/serverlessworkflow/main/cloud/build-workflow-image-with-quarkus-cli.html#proc-building-serverless-workflow-application

"Step 3: alternatively, you can use a Maven profile to..."

I think you can consider taking this "procedure" approach here too since you're describing a sequence of steps to achieve a certain goal.